addons/crm/models/crm_team.py

```diff
@@ -463,7 +473,17 @@ def _allocate_leads(self, creation_delta_days=7):
 
             # assign + deduplicate and concatenate results in teams_data to keep some history
             candidate_lead = teams_data[team]["leads"][0]
-            assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            try:
+                assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            except Exception as e:
+                if auto_commit:
+                    _logger.error("[Assign Lead] Error while assigning lead %s (ID: %d) to team %s (ID: %d). Error: %s", candidate_lead.name, candidate_lead.id, team.name, team.id, str(e))
+                    _logger.error("[Assign Lead] %d leads (IDs: %s) assigned to %s team (ID: %d) before crash.", len(teams_data[team]["assigned"]), list(teams_data[team]["assigned"]), team.name, team.id)
```

Comment:
Would remove this second one as it won't necessarily be accurate (not necessarily committed by this point)